### PR TITLE
fix: native union changes

### PR DESCRIPTION
**Pull request by rgraff**
This fixes an issue that I ran into while writing edge case tests for ash_paper_trail's full diff feature.  Updating a union value to a native type (:string) error will error when calling `prepare_change` and `handle_change`.

```
     ** (UndefinedFunctionError) function :string.prepare_change/3 is undefined or private
     code: ctx.resource.update!(res, %{
     stacktrace:
       (stdlib 5.1.1) :string.prepare_change(nil, "https://www.just-a-link.com", [])
       (ash 2.15.8) lib/ash/type/union.ex:589: Ash.Type.Union.do_prepare_change/4
       (ash 2.15.8) lib/ash/changeset/changeset.ex:3433: Ash.Changeset.change_attribute/3
       (stdlib 5.1.1) maps.erl:416: :maps.fold_1/4
       (ash 2.15.8) lib/ash/changeset/changeset.ex:913: Ash.Changeset.prepare_changeset_for_action/4
       (ash 2.15.8) lib/ash/changeset/changeset.ex:875: Ash.Changeset.do_for_action/4
       (ash_paper_trail 0.1.0) deps/ash/lib/ash/code_interface.ex:562: AshPaperTrail.Test.Posts.Page.update!/3
```

Discussed briefly in Discord: https://discord.com/channels/711271361523351632/955933117573652563/1164034328548884480

# Contributor checklist

- [x] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [ ] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

File: lib/ash/type/union.ex [hunk 1]
```diff
@@ -521,8 +521,12 @@ defmodule Ash.Type.Union do
          {:ok, type_config} <- Keyword.fetch(type_configs, type_name),
          {:ok, type} <- Keyword.fetch(type_config, :type),
          type_constraints <- Keyword.get(type_config, :constraints, []),
-         {:ok, new_value} <- type.handle_change(nil, new_value, type_constraints) do
-      {:ok, %Ash.Union{type: type_name, value: new_value}}
+         true <- :erlang.function_exported(type, :handle_change, 3),
```
Comment (zachdaniel) [inline lib/ash/type/union.ex:524]: I think what we need here instead is `type <- Ash.Type.get_type(type)`, which will convert `:string` to `Ash.Type.String`.

Comment (rgraff) [reply lib/ash/type/union.ex]: @zachdaniel resolved

**Review by zachdaniel — APPROVED** [review]
